liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,404 @@
+      .dividedBy(tokens[tokenAddress].cTokenDecimalsMult)
+      .dividedBy(100); // Not sure where this 100 comes from I didn't see it in the docs
+
+    /* eslint-disable no-param-reassign */
+    if (data.borrow[tokenAddress] === undefined) data.borrow[tokenAddress] = {};
+    if (data.supply[tokenAddress] === undefined) data.supply[tokenAddress] = {};
+    /* eslint-enable no-param-reassign */
+  }
+}
+// #endregion
+
+// Initializes data required for handler
+function provideInitialize(data) {
+  return async function initialize() {
+    // #region Assign configurable fields
+    /* eslint-disable no-param-reassign */
+    data.alertMinimumIntervalSeconds = config.alertMinimumIntervalSeconds;
+    data.protocolName = config.protocolName;
+    data.protocolAbbreviation = config.protocolAbbreviation;
+    data.developerAbbreviation = config.developerAbbreviation;
+    data.alert = config.liquidationMonitor.alert;
+    data.minimumLiquidationInUSD = config.liquidationMonitor.triggerLevels.minimumLiquidationInUSD;
+    data.lowHealthThreshold = config.liquidationMonitor.triggerLevels.lowHealthThreshold;
+    data.cTokenABI = getAbi('cErc20.json');
+    data.provider = getEthersProvider();
+    data.accounts = {}; // Health of all accounts, calcHealth, lastUpdated, [assetsIn addresses]?
+    data.supply = {}; // qty of cTokens (not Tokens)
+    data.borrow = {}; // qty of Tokens (not cTokens)
+    data.tokens = {}; // each cToken's address, symbol, contract, ratio, price, lastUpdatePrice
+    data.newAccounts = []; // New account from transaction events
+    data.totalNewAccounts = 0;
+
+    // Compound API filter and Comptroller contract
+    const {
+      comptrollerAddress,
+      maxTrackedAccounts,
+      oracleAddress,
+      oneInchAddress,
+      triggerLevels: {
+        maximumHealth,
+        minimumBorrowInETH,
+      },
+    } = config.liquidationMonitor;
+    const comptrollerABI = getAbi(config.liquidationMonitor.comptrollerABI);
+    const oracleABI = getAbi(config.liquidationMonitor.oracleABI);
+    const oneInchABI = getAbi(config.liquidationMonitor.oneInchABI);
+    data.comptrollerAddress = comptrollerAddress;
+    data.comptrollerContract = new ethers.Contract(
+      comptrollerAddress,
+      comptrollerABI,
+      data.provider,
+    );
+    data.oneInchContract = new ethers.Contract(
+      oneInchAddress,
+      oneInchABI,
+      data.provider,
+    );
+    // Unused but available for the future
+    data.oracleContract = new ethers.Contract(
+      oracleAddress,
+      oracleABI,
+      data.provider,
+    );
+    /* eslint-enable no-param-reassign */
+    // #endregion
+
+    // #region Get initial accounts from Compound API
+    // Find total number of results with the first request
+    const initialRequest = buildJsonRequest(maximumHealth, minimumBorrowInETH, 1, 1);
+    const initialResults = await callCompoundAPI(initialRequest);
+    const totalEntries = initialResults.pagination_summary.total_entries;
+
```

Comment:
There's a bit of a race condition here. If you grab total pages and then iterate assuming total pages never changes, and transactions are made mid page request loop, things get a little squirrely.. 
Unfortunately, this form of pagination is just messy if not done very carefully, this is not on you. Hopefully new transactions get added to the end of the list, otherwise you may wind up with duplicates.

The only vaguely ok way to do this is just to loop over the stuff in L157->L162 until the callCompoundAPI request fails due to an out-of-range, and hope you got a coherent snapshot. You can then get/use the length of your foundAccounts array if you need a max length from there.